src/comment.rs

```diff
@@ -446,36 +448,47 @@ struct ItemizedBlock {
 }
 
 impl ItemizedBlock {
-    /// Returns `true` if the line is formatted as an item
-    fn is_itemized_line(line: &str) -> bool {
-        let trimmed = line.trim_start();
-        trimmed.starts_with("* ") || trimmed.starts_with("- ") || trimmed.starts_with("> ")
-    }
-
-    /// Creates a new ItemizedBlock described with the given line.
-    /// The `is_itemized_line` needs to be called first.
-    fn new(line: &str) -> ItemizedBlock {
-        let space_to_sigil = line.chars().take_while(|c| c.is_whitespace()).count();
-        // +2 = '* ', which will add the appropriate amount of whitespace to keep itemized
-        // content formatted correctly.
-        let mut indent = space_to_sigil + 2;
-        let mut line_start = " ".repeat(indent);
-
-        // Markdown blockquote start with a "> "
-        if line.trim_start().starts_with(">") {
-            // remove the original +2 indent because there might be multiple nested block quotes
-            // and it's easier to reason about the final indent by just taking the length
-            // of th new line_start. We update the indent because it effects the max width
-            // of each formatted line.
-            line_start = itemized_block_quote_start(line, line_start, 2);
-            indent = line_start.len();
+    /// Returns the sigil's (e.g. "- ", "* ", or "1. ") length or None if there is no sigil.
+    fn get_sigil_length(trimmed: &str) -> Option<usize> {
+        if trimmed.starts_with("* ") || trimmed.starts_with("- ") || trimmed.starts_with("> ") {
+            return Some(2);
         }
-        ItemizedBlock {
-            lines: vec![line[indent..].to_string()],
-            indent,
-            opener: line[..indent].to_string(),
-            line_start,
+
+        for suffix in [". ", ") "] {
+            if let Some((prefix, _)) = trimmed.split_once(suffix) {
+                if prefix.len() <= 2 && prefix.chars().all(|c| char::is_ascii_digit(&c)) {
+                    return Some(prefix.len() + suffix.len());
+                }
+            }
```

Comment:
I see the test cases and the comments that were added there, but can we also include a clarifying comment here to fully explain the rational for why `prefix.len() <= 2` was chosen.

Reply:
> I see the test cases and the comments that were added there, but can we also include a clarifying comment here to fully explain the rational for why prefix.len() <= 2 was chosen.

Done.

> It may even be helpful to factor this loop into its own function (something like get_numbered_sigils) 

I think it's nice to keep this in a single function, because:
- the doc comments apply to both the old-styke and new-style sigils/markers
- the hypothetical new function would only have a single caller (unlike the new `get_marker_length` which has 2 callers)

> to make it more obvious that the following is used to determine numbered markdown lists.

Done, I hope?  The doc comment + comments in the body of the function hopefully help with that?